### PR TITLE
HGSS Style Pokedex

### PR DESCRIPTION
**Pull request by AaghatIsLive**
## Description
This PR adds @TheXaman's HGSS Style Pokedex

## Images
pokedex_plus_hgss

## **Discord contact info**
aaghat

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Note for the reviewer: ***DON'T*** squash this one to avoid conflicts with users that already use this feature branch.

**Comment by AaghatIsLive** [comment]
Ready for Review!

File: include/daycare.h [hunk 1]
```diff
@@ -30,5 +30,9 @@ void SetDaycareCompatibilityString(void);
 bool8 NameHasGenderSymbol(const u8 *name, u8 genderRatio);
 void ShowDaycareLevelMenu(void);
 void ChooseSendDaycareMon(void);
+#if POKEDEX_PLUS_HGSS == TRUE
+u8 GetEggMovesSpecies(u16 species, u16 *eggMoves);
+bool8 SpeciesCanLearnEggMove(u16 species, u16 move);
+#endif
```
Comment (AsparagusEduardo) [inline include/daycare.h:36]: These 2 functions might be useful to have available, regardless of the `POKEDEX_PLUS_HGSS` config, so I would remove the ifdefs here and in the definitions.

File: src/fonts.c [hunk 2]
```diff
@@ -37,6 +37,42 @@ ALIGNED(4) const u8 gFontSmallNarrowLatinGlyphWidths[] = {
 };
 
 ALIGNED(4) const u16 gFontSmallLatinGlyphs[] = INCBIN_U16("graphics/fonts/small.latfont");
+#if POKEDEX_PLUS_HGSS == TRUE
```
Comment (AsparagusEduardo) [inline src/fonts.c:40]: Since we're adding the Egg icon to the font itself, I would just modify the specific width.

File: src/graphics.c [hunk 3]
```diff
@@ -1739,6 +1739,41 @@ const u16 gPokedexSearchMenu_Pal[] = INCBIN_U16("graphics/pokedex/search_menu.gb
 const u32 gPokedexSearchMenu_Gfx[] = INCBIN_U32("graphics/pokedex/search_menu.4bpp.lz");
 const u32 gPokedexSearchMenuNational_Tilemap[] = INCBIN_U32("graphics/pokedex/search_menu_national.bin.lz");
 const u32 gPokedexSearchMenuHoenn_Tilemap[] = INCBIN_U32("graphics/pokedex/search_menu_hoenn.bin.lz");
+// Pokedex HGSS
+#if POKEDEX_PLUS_HGSS == TRUE
+const u16 gPokedexPlusHGSS_Default_Pal[]             = INCBIN_U16("graphics/pokedex/hgss/HGSS_palette_default.gbapal");
```
Comment (AsparagusEduardo) [inline src/graphics.c:1744]: Alignment here is weird. Let's remove it to keep consistant with previous graphics.

File: src/strings.c [hunk 4]
```diff
@@ -1833,3 +1833,131 @@ const u8 gText_Berry[] = _("BERRY");
 const u8 gText_Berries[] = _("BERRIES");
 const u8 gText_ExpShareOn[] = _("The Exp. Share has been turned on.{PAUSE_UNTIL_PRESS}");
 const u8 gText_ExpShareOff[] = _("The Exp. Share has been turned off.{PAUSE_UNTIL_PRESS}");
+
+//HGSS_Ui PokedexPlus
+#if POKEDEX_PLUS_HGSS == TRUE
+const u8 gText_Stats_Buttons[]  = _("{A_BUTTON}TOGGLE   {DPAD_UPDOWN}MOVES");
```
Comment (AsparagusEduardo) [inline src/strings.c:1839]: Remove alignment to keep consistency with previous strings.

File: src/pokedex_plus_hgss.c [hunk 5]
```diff
@@ -0,0 +1,8542 @@
...
+    bool8 justScrolled; //HGSS_Ui
+    u8 splitIconSpriteId;  //HGSS_Ui Physical/Special Split
+    u8 numEggMoves; //HGSS_Ui
+    u8 numLevelUpMoves; //HGSS_Ui
+    u8 numTMHMMoves; //HGSS_Ui
+    u8 numTutorMoves; //HGSS_Ui
+    u8 numPreEvolutions; //HGSS_Ui
+    struct PokemonStats sPokemonStats; //HGSS_Ui
+    struct EvoScreenData sEvoScreenData; //HGSS_Ui
+    struct FromScreenData sFormScreenData; //HGSS_Ui
```
Comment (AsparagusEduardo) [inline src/pokedex_plus_hgss.c:285]: Remove the unneeded `HGSS_Ui` comments.

File: src/pokedex.c [hunk 6]
```diff
@@ -14,6 +14,9 @@
 #include "overworld.h"
 #include "palette.h"
 #include "pokedex.h"
+#if POKEDEX_PLUS_HGSS == TRUE
+#include "pokedex_plus_hgss.h"
```
Comment (AsparagusEduardo) [inline src/pokedex.c:18]: `include/pokedex_plus_hgss.h` is already cleared out when `POKEDEX_PLUS_HGSS` is false, so there's no need to hide this include as well.

File: src/pokedex_area_screen.c [hunk 7]
```diff
@@ -666,19 +667,55 @@ static void Task_HandlePokedexAreaScreenInput(u8 taskId)
             return;
         break;
     case 1:
+#if POKEDEX_PLUS_HGSS == TRUE
         if (JOY_NEW(B_BUTTON))
```
Comment (AsparagusEduardo) [inline src/pokedex_area_screen.c:671]: There's no differences between the 2 settings here.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Initial review.

**Comment by AaghatIsLive** [comment]
Applied requested changes.

File: graphics/pokedex/hgss/RES_HGSS_palette.pal
Comment (AsparagusEduardo) [inline graphics/pokedex/hgss/RES_HGSS_palette.pal:1]: Is this unused?

File: src/pokemon_summary_screen.c [hunk 8]
```diff
@@ -935,13 +935,13 @@ static const union AnimCmd *const sSpriteAnimTable_MoveTypes[NUMBER_OF_MON_TYPES
     sSpriteAnim_CategoryTough,
 };
 
-static const struct CompressedSpriteSheet sSpriteSheet_MoveTypes =
+const struct CompressedSpriteSheet sSpriteSheet_MoveTypes =
```
Comment (AsparagusEduardo) [inline src/pokemon_summary_screen.c:938]:
```suggestion
const struct CompressedSpriteSheet gSpriteSheet_MoveTypes =
```

File: src/pokemon_summary_screen.c [hunk 9]
```diff
@@ -935,13 +935,13 @@ static const union AnimCmd *const sSpriteAnimTable_MoveTypes[NUMBER_OF_MON_TYPES
...
 {
     .data = gMoveTypes_Gfx,
     .size = (NUMBER_OF_MON_TYPES + CONTEST_CATEGORIES_COUNT) * 0x100,
     .tag = TAG_MOVE_TYPES
 };
-static const struct SpriteTemplate sSpriteTemplate_MoveTypes =
+const struct SpriteTemplate sSpriteTemplate_MoveTypes =
```
Comment (AsparagusEduardo) [inline src/pokemon_summary_screen.c:944]:
```suggestion
const struct SpriteTemplate gSpriteTemplate_MoveTypes =
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Comment by AsparagusEduardo** [comment]
Oh and one more thing, in the Discord we discussed adding special handling for Mew so that it shows a single entry that says "ALL" for TMs and Tutors. Could you add this?

**Comment by AsparagusEduardo** [comment]
> Oh and one more thing, in the Discord we discussed adding special handling for Mew so that it shows a single entry that says "ALL" for TMs and Tutors. Could you add this?

If not, I'll try to PR it myself.

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**